alibi_detect/cd/mmd.py

```diff
@@ -100,9 +108,21 @@ def __init__(
 
         if backend == 'tensorflow' and has_tensorflow:
             kwargs.pop('device', None)
-            self._detector = MMDDriftTF(*args, **kwargs)  # type: ignore
+            if estimator == 'quad':
+                self._detector = MMDDriftTF(*args, **kwargs)  # type: ignore
+            elif estimator == 'linear':
+                kwargs.pop('n_permutations', None)
+                self._detector = LinearTimeMMDDriftTF(*args, **kwargs)  # type: ignore
```

Comment:
Since the logic to set `self._detector` is located here, we should add additional tests to `alibi_detect/cd/tests/test_mmd.py` to check that the correct subclass is selected conditional on `backend` and `estimator`.

Reply:
Indeed, will modify the tests.

Reply:
Simply rewrite the test to go through different `backend` and `estimator` options, should do the job.